New blog: Best open source AB testing tools (#3472)
PostHog/posthog.com: contents/blog/best-open-source-ab-testing-tools.md

---
date: 2022-05-24
title: The 5 best free and open-source A/B testing tools
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
featuredImage: ../images/blog/experiments.jpeg
featuredImageType: full
categories: ["Open source", "Guides"]
author: ["hanna-crombie"]
---

Also known as split testing or multivariate testing, A/B testing is the implementation of experiments which split your audience to test variations of a product design, new feature, call to action, landing page – anything you can imagine, really.

Here, we’re sharing our guide to the best free and [open-source tools](/blog/best-open-source-analytics-tools) for experimenting with A/B tests because the best things in life are free (and open source, obviously).

## 1. [PostHog](https://posthog.com)

![PostHog - best open source ab testing tools](../images/blog/open-source-testing-tools/img1_PostHog.png)

PostHog is an [all-in-one product analytics suite](/product) that integrates all of the essential features you need to understand your users. It collects and visualizes data on how people are using your products, provides insights into trends and retention, and helps to remove bottlenecks and reduce churn.

> Our A/B testing capabilities, which we call our Experimentation Suite, is complemented by other tools like [Feature Flags](/product/feature-flags), [Session Recording](/prodcut/session-recording), [Heatmaps](/product/heatmaps), [User Paths](/product/user-paths) and more, making it one of the most comprehensive analytics platforms around – open source, or otherwise.

Author: ```suggestion
Our A/B testing features, which we call our Experimentation suite, are complemented by other tools like [Feature Flags](/product/feature-flags), [Session Recording](/prodcut/session-recording), [Heatmaps](/product/heatmaps), [User Paths](/product/user-paths) and more, making it one of the most comprehensive analytics platforms around – open source, or otherwise.
```

[PostHog’s Experimentation suite](https://posthog.com/docs/user-guides/experimentation) allows you to flexibly test your hypotheses with up to three test variants. A control group is also assigned and PostHog will automatically estimate the number of users exposed, and we use Bayesian analysis to calculate the results and ensure they're significant.

### Who is PostHog for?

> PostHog is a great choice for any team that wants a complete view how people use their product. Not only is the testing functionality comprehensive, it’s tightly integrated with a complete suite of product analytics tools – you don’t need any external tools. You can also [self-host PostHog](/signup/self-host), so data never leaves your infrastructure.

Author: ```suggestion
PostHog is a great choice for any team that wants a complete view how people use their product. Not only is the testing functionality comprehensive, it’s tightly integrated with a complete product analytics suite – you don’t need any external tools. You can also [self-host PostHog](/signup/self-host), so data never leaves your infrastructure.
```

### Strengths

- Full product analytics suite
- Easy to get set up – no SQL necessary
- Ability to select segments for experiments
- Feature flags for incremental roll outs
- Event pipelines for integration with data warehouses
- Traffic visualization with user path analysis
- Self-hosted and cloud-hosted options available

### How much does it cost?

> We’re confident PostHog is the number one choice here, but if you want to test our theory you can [try our self-serve plans for free](/pricing). 

Author: ```suggestion
We’re confident PostHog is the number one choice here, but you can decide for youself via [our free self-serve plans](/pricing). 
```

If you want to self-host, you can use Experimentation with our Scale plan, but you can also use PostHog Cloud if you prefer a fully-managed option with no infrastructure to deploy. 

> Both PostHog Scale and Cloud are [free up to 1 million events per month](/pricing).

Author: ```suggestion
Though Experimentation isn't part of the free-for-life open source release, it's [free to use up to 1 million events per month](/pricing) on PostHog's core paid plans – Scale for self-hosting, or PostHog Cloud if you prefer fully-managed experience.
```

> We’re confident PostHog is the number one choice here, but you can decide for yourself via [our free self-serve plans](/pricing). 
> 
> If you want to self-host, you can use Experimentation with our Scale plan, but you can also use PostHog Cloud if you prefer a fully-managed option with no infrastructure to deploy. 
> 
> Both PostHog Scale and Cloud are [free up to 1 million events per month](/pricing).

Author: ```suggestion
While Experimentation isn't part of the free-for-life open source release, it's [free to use up to 1 million events per month](/pricing) on PostHog's paid plans – Scale for self-hosted, and PostHog Cloud if your a prefer a fully-managed SaaS experience.
```

### Open source license

> PostHog Open Source is available under an MIT license. You can [see further details on GitHub]. Experimentation isn't part of our open source release as it requires a Scale license, but Scale is free to use up to 1 million events per month and you can set event limits to ensure you don’t go over. Read our docs for more on [how to self-host PostHog](https://posthog.com/docs/self-host).

Author: ```suggestion
PostHog Open Source is available under an [MIT license](https://github.com/PostHog/posthog). Experimentation isn't part of our open source release as it requires a Scale license, but Scale is free to use up to 1 million events per month and you can set event limits to ensure you don’t go over. Read our docs for more on [how to self-host PostHog](https://posthog.com/docs/self-host).
```

<ArrayCTA />

**Further reading:**
- [The best open source Hotjar and FullStory alternatives](/blog/best-open-source-session-replay-tools)
- [What we've learned about running effective A/B tests](/blog/experiments)
- [The differences between PostHog and Amplitude](/blog/posthog-vs-amplitude)

## 2. [GrowthBook](https://www.growthbook.io/)

![GrowthBook - best open source ab testing tools](../images/blog/open-source-testing-tools/growthbook.png)

> GrowthBook is an open-source platform for feature flags and A/B tests which helps teams to deploy code efficiently and analyze experiments. Built by engineers who wanted better adoption insights into their releases, it is a modular solution that promotes feature flagging as an essential step in the development process and can be used as a full-stack platform, a plugin feature flagging tool or an analysis engine.

Author: ```suggestion
GrowthBook is an open-source platform for feature flags and A/B tests which helps teams to deploy code efficiently and analyze experiments. Built by engineers who wanted better insights on new releases, it is a modular solution that promotes feature flagging as an essential step in the development process and can be used as a full-stack platform, a plugin feature flagging tool or an analysis engine.
```

### Who is GrowthBook for?

GrowthBook is built for data, engineering and product teams who need the power of a customizable platform without having to build it in house.

### Strengths

- Modular platform to adapt to your needs
- Lightweight SDKs for speed
- Easy to implement – no need for data or engineering resources
- Self-hosted and managed options available

### How much does it cost?

> GrowthBook open source is free and includes unlimited experiments. A hosted version includes a visual A/B test editor and is free up to 5 users, with pricing at $20 per user per month after.

Author: ```suggestion
GrowthBook open source is free and includes unlimited experiments. A hosted version includes a visual A/B test editor and is free up to 5 users, with pricing at $20 per user per month thereafter.
```

### Open sources license

GrowthBook is available under an MIT license. Visit the [Growthbook GitHub repo](https://github.com/growthbook/growthbook/) for more info. 

## 3. [Unleash](https://www.getunleash.io/)

![Unleash - best open source ab testing tools](../images/blog/open-source-testing-tools/img2_Unleash.png)

Unleash is a feature management platform which provides an overview of all features across your applications and services. The platform empowers engineering teams to implement A/B tests via feature toggles and offers accurate user targeting.

> You can use Unleash to define the rules of your experiments, however the platform doesn’t provide all the tools you need to manage A/B tests end-to-end and you will need to connect your experiment with an analytics platform like Google Analytics or Mixpanel in order to visualize results.

Author: ```suggestion
You can use Unleash to define the rules of your experiments, but the platform doesn’t provide all the tools you need to manage A/B tests end-to-end. Instead, you'll need to connect your experiment with an analytics platform like Google Analytics or Mixpanel in order to visualize results.
```

---
date: 2022-05-24
title: The 5 best free and open-source A/B testing tools
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
featuredImage: ../images/blog/experiments.jpeg
featuredImageType: full
categories: ["Open source", "Guides"]
author: ["hanna-crombie"]
---

Developers and marketers are customer experts, so it’s tempting to assume we can instinctively make the best decisions for user engagement. This is a trap.
 
Leaders don’t want to make changes without evidence and however much we think we know about our users, making changes on theory alone is fraught with danger. That’s why you need an A/B testing tool.

> Also known as split testing or multivariate testing, A/B testing is the implementation of experiments which split your audience to test variations of a product design, new feature, call to action, landing page – anything you can imagine, really.

Reviewer: I would personally cut the first two paras and just open with the third one ('Also known as split testing....'). The first two paras don't add any value, and anyone searching for this already will be convinced enough that they need an A/B testing tool.

Author: Fair. This is already considerably cut down from the original verision, but I'm always in favour of jumping right in.

Here, we’re sharing our guide to the best free and [open-source tools](/blog/best-open-source-analytics-tools) for experimenting with A/B tests because the best things in life are free (and open source, obviously).

## 1. [PostHog](https://posthog.com)

![PostHog - best open source ab testing tools](../images/blog/open-source-testing-tools/img1_PostHog.png)

PostHog is an [all-in-one product analytics suite](/product) that integrates all of the essential features you need to understand your users. It collects and visualizes data on how people are using your products, provides insights into trends and retention, and helps to remove bottlenecks and reduce churn.

> Our A/B testing capabilities, which we call our Experimentation Suite, is complemented by other tools like [Feature Flags](/product/feature-flags), [Session Recording](/prodcut/session-recording), [Heatmaps](/product/heatmaps), [User Paths](/product/user-paths) and more, making it one of the most comprehensive analytics platforms around – open source, or otherwise.

Reviewer: Our A/B testing capabilities.... _are_ complemented by other tools?

[PostHog’s Experimentation suite](https://posthog.com/docs/user-guides/experimentation) allows you to flexibly test your hypotheses with up to three test variants. A control group is also assigned and PostHog will automatically estimate the number of users exposed, and we use Bayesian analysis to calculate the results and ensure they're significant.

### Who is PostHog for?

> PostHog is a great choice for any team that wants a complete view how people use their product. Not only is the testing functionality comprehensive, it’s tightly integrated with a complete suite of product analytics tools – you don’t need any external tools. You can also [self-host PostHog](/signup/self-host), so data never leaves your infrastructure.

Reviewer: Too many 'tools'

> PostHog is a great choice for any team that wants a complete view how people use their product. Not only is the testing functionality comprehensive, it’s tightly integrated with a complete suite of product analytics tools – you don’t need any external tools. You can also [self-host PostHog](/signup/self-host), so data never leaves your infrastructure.

Reviewer 2: ```suggestion
PostHog is a great choice for any team that wants a complete view of how people use their product. Not only is the testing functionality comprehensive, it’s tightly integrated with a complete suite of product analytics tools – you don’t need any external tools. You can also [self-host PostHog](/signup/self-host), so data never leaves your infrastructure.
```

### Strengths

- Full product analytics suite
- Easy to get set up – no SQL necessary
- Ability to select segments for experiments
- Feature flags for incremental roll outs
- Event pipelines for integration with data warehouses
- Traffic visualization with user path analysis
- Self-hosted and cloud-hosted options available

### How much does it cost?

> We’re confident PostHog is the number one choice here, but if you want to test our theory you can [try our self-serve plans for free](/pricing). 

Reviewer: 'If you want to test our theory' sounds weirdly clunky

If you want to self-host, you can use Experimentation with our Scale plan, but you can also use PostHog Cloud if you prefer a fully-managed option with no infrastructure to deploy. 

Both PostHog Scale and Cloud are [free up to 1 million events per month](/pricing).

### Open source license

> PostHog Open Source is available under an MIT license. You can [see further details on GitHub]. Experimentation isn't part of our open source release as it requires a Scale license, but Scale is free to use up to 1 million events per month and you can set event limits to ensure you don’t go over. Read our docs for more on [how to self-host PostHog](https://posthog.com/docs/self-host).

Reviewer: [see further details on GitHub] - missing link

<ArrayCTA />

**Further reading:**
- [The best open source Hotjar and FullStory alternatives](/blog/best-open-source-session-replay-tools)
- [What we've learned about running effective A/B tests](/blog/experiments)
- [The differences between PostHog and Amplitude](/blog/posthog-vs-amplitude)

## 2. [GrowthBook](https://www.growthbook.io/)

![GrowthBook - best open source ab testing tools](../images/blog/open-source-testing-tools/growthbook.png)

> GrowthBook is an open-source platform for feature flags and A/B tests which helps teams to deploy code efficiently and analyze experiments. Built by engineers who wanted better adoption insights into their releases, it is a modular solution that promotes feature flagging as an essential step in the development process and can be used as a full-stack platform, a plugin feature flagging tool or an analysis engine.

Reviewer: Reads weirdly - 'Built by engineers who wanted better insights into adoption of their releases, etc.' maybe?

### Who is GrowthBook for?

GrowthBook is built for data, engineering and product teams who need the power of a customizable platform without having to build it in house.

### Strengths

- Modular platform to adapt to your needs
- Lightweight SDKs for speed
- Easy to implement – no need for data or engineering resources
- Self-hosted and managed options available

### How much does it cost?

> GrowthBook open source is free and includes unlimited experiments. A hosted version includes a visual A/B test editor and is free up to 5 users, with pricing at $20 per user per month after.

Reviewer: 'after' -> 'beyond that'

### Open sources license

GrowthBook is available under an MIT license. Visit the [Growthbook GitHub repo](https://github.com/growthbook/growthbook/) for more info. 

## 3. [Unleash](https://www.getunleash.io/)

![Unleash - best open source ab testing tools](../images/blog/open-source-testing-tools/img2_Unleash.png)

Unleash is a feature management platform which provides an overview of all features across your applications and services. The platform empowers engineering teams to implement A/B tests via feature toggles and offers accurate user targeting.

> You can use Unleash to define the rules of your experiments, however the platform doesn’t provide all the tools you need to manage A/B tests end-to-end and you will need to connect your experiment with an analytics platform like Google Analytics or Mixpanel in order to visualize results.

Reviewer: This sentence could do with breaking up, it runs on weirdly.

### Who is Unleash for?

> Unleash is for autonomous development teams who want to be free from constraints and get on with testing and validating new features safely. They build with key developer concerns in mind: integration, privacy, resilience and performance.

Reviewer: 'who want to be free from constraints' sounds really non-specific. This section almost reads like it's been copied and pasted from Unleash's marketing materials.

### Open source license

The open source version is available under an Apache license. Visit the [Unleash GitHub repo](https://github.com/Unleash/unleash) for more info.

## 4. [Mojito](https://mojito.mx/)

![Unleash - best open source ab testing tools](../images/blog/open-source-testing-tools/img3_Mojito.png)

Mojito is an open split testing stack that lets you build, launch and analyze experiments via Git/CI. The stripped-down tool consists of three modular components – a front-end library for bucketing users and tracking them, data models and events for fast reporting, and reporting templates and functions so you can build your own visualizations for experiment analysis.

### Who is Mojito for?

Mojito is a fully source-controlled stack for developers and technical product teams who want a simple solution with unlimited customisation.

### Strengths

- Simple APIs and unlimited customisation
- No need for 3rd party analytics
- Error tracking and handling
- Lightweight solution for optimized page load speed

### How much does it cost?

> Mojito is fully free and open source and you don’t need an account to get started.

Reviewer: '...open source**,** and you don't need...'
